framework/mongo/validators.py

```diff
@@ -1,9 +1,41 @@
 """
 Validators for use in ModularODM
 """
-from modularodm.exceptions import ValidationValueError
+import collections
+from modularodm.exceptions import ValidationError, ValidationValueError
 
 
 def string_required(value):
     if value is None or value == '':
         raise ValidationValueError('Value must not be empty.')
+    return True
+
+
+def choice_in(choices, ignore_case=False):
+    """
+    Validate that the option provided is one of a restricted set of choices
+
+    Returns a callable that can be used as a validator in ModularODM
+    :param choices: An iterable of choices allowed for use
+    :param bool ignore_case: If True, perform case-insensitive comparison (for strings) and otherwise match as-is
+    :return:
+    """
+
+    if ignore_case:
+        choice_set = frozenset(e.upper() if isinstance(e, basestring) else e
+                               for e in choices)
+    else:
+        choice_set = frozenset(choices)
+
+    def validator(value):
+
+        if not isinstance(value, collections.Hashable):
```

Comment:
I think EAFP (try-except) is probably preferable in this situation, but this is fine, too; `Hashable` does the `__subclass__` magic we discovered a while back.